packages/xstate-wallet/src/chain.ts

```diff
@@ -237,39 +236,57 @@ export class ChainWatcher implements Chain {
   private _adjudicator?: Contract;
   private _assetHolders: Contract[];
   private mySelectedAddress: string | null;
+  private provider: ReturnType<typeof getProvider>;
+  private get signer() {
+    if (typeof this.selectedAddress !== 'string') throw 'Ethereum not enabled';
+
+    return this.provider.getSigner(this.selectedAddress);
+  }
 
   public async initialize() {
-    const provider = getProvider();
-    const signer = provider.getSigner();
+    this.provider = getProvider();
```

Comment:
I think we should make this a constructor argument of `Chain`

Reply:
https://github.com/statechannels/monorepo/issues/1394